### PR TITLE
[Components with more than 1 service] LPS-192654 Refactor SXPParameterDataCreator component

### PR DESCRIPTION
**Pull request by RenanUmV**
[LPS-192654](https://liferay.atlassian.net/browse/LPS-192654)

### CODE REVIEW

**Comment by RenanUmV** [comment]
ci:test:help

**Comment by RenanUmV** [comment]
ci:test:search

File: modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/blueprint/parameter/SXPParameterDataCreator.java [hunk 1]
```diff
@@ -104,26 +93,8 @@ public SXPParameterData create(
...
-				_sxpParameterContributors) {
-
-			sxpParameterContributorDefinitions.addAll(
-				sxpParameterContributor.getSXPParameterContributorDefinitions(
-					companyId, locale));
-		}
-
-		return sxpParameterContributorDefinitions;
+	public SXPParameterContributor[] getSxpParameterContributors() {
+		return _sxpParameterContributors;
```
Comment (rafaprax) [inline modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/blueprint/parameter/SXPParameterDataCreator.java:97]: Hey @RenanUmV, this method seems off into the `SXPParameterDataCreator`. Let's try this:

1.  extract this logic to a `SXPParameterContributorProvider` class and call it into the `SXPParameterContributorDefinitionProviderImpl`. 
2. Then rename the `SXPParameterDataCreator` to `SXPParameterDataCreatorUtil` and make this a simple Util class.
3. Pass the `SXPParameterContributorProvider` as a parameter to the `SXPParameterDataCreatorUtil.create` method

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by rafaprax** [comment]
See my comments and resend the fix. Thx

**Pull request closed without merging**